src/App/index.jsx

```diff
@@ -82,7 +82,8 @@ class App extends React.Component {
           {authReady ? (
             <AuthContext.Provider value={this.authController}>
               <Switch>
-                <PropsRoute path="/" exact component={Main} />
+                <PropsRoute path="/:view" component={Main} />
```

Comment:
The `Main` component handles getting all of the data for each of the views. The data is passed to those components from the `Main` state.

Reply:
From @helfi92 :
> You can avoid
> 
> ```js-jsx
> {view === ... && ()}
> {view === ... && ()}
> {view === ... && ()}
> ...
> ```
> 
> by doing [`<PropsRoute path="/teams" component={Teams} />`] (edited)

Reply:
How do you suggest I proceed?
Can I use `Route` or `Switch` within the `Main` component?

Reply:
Something like:

```jsx
// index.jsx
<Header /> 
<Main />
```

```jsx
// Main.jsx
<Switch>
  <PropsRoute path="/teams" component={Teams} />
  <PropsRoute path="/components" component={Components} />
  <PropsRoute path="/" component={Reportees} />
</Switch>
```

>  Can I use Route or Switch within the Main component?

You can use Route or Switch in any component you want but I wouldn't suggest having more than a single Switch statement for the current app unless the site starts having too many routes to keep track of in one place.

Reply:
Where do you handle using React.lazy & <Suspense> for that?

Reply:
I have this (w/o React.lazy, otherwise, I get a bunch of errors):
```diff
diff --git a/src/views/Main/index.jsx b/src/views/Main/index.jsx
index 4e1d0b6..ff50624 100644
--- a/src/views/Main/index.jsx
+++ b/src/views/Main/index.jsx
@@ -1,5 +1,8 @@
 import React, { Component, Suspense } from 'react';
+import { Switch } from 'react-router-dom';
+import PropsRoute from '../../components/PropsRoute';
 import AuthContext from '../../components/auth/AuthContext';
 import Header from '../../components/Header';
+import Reportees from '../../components/Reportees';
 import getAllReportees from '../../utils/getAllReportees';
 import getBugzillaOwners from '../../utils/getBugzillaOwners';
@@ -11,5 +14,4 @@ const BugzillaComponents = React.lazy(() => import('../../components/BugzillaCom
 const BugzillaComponentDetails = React.lazy(() => import('../../components/BugzillaComponentDetails'));
 const PersonDetails = React.lazy(() => import('../../components/PersonDetails'));
-const Reportees = React.lazy(() => import('../../components/Reportees'));

 const DEFAULT_STATE = {
@@ -235,4 +237,15 @@ class MainContainer extends Component {
             </Suspense>
           )}
+          <Switch>
+            {partialOrg && (
+              <PropsRoute
+                path="/"
+                component={Reportees}
+                ldapEmail={ldapEmail}
+                partialOrg={partialOrg}
+                onPersonDetails={this.handleShowPersonDetails}
+              />
+            )}
+          </Switch>
           {view === 'reportees' && partialOrg && (
             <Suspense fallback={<div>Loading...</div>}>
```

Reply:
I'm not sure. Maybe because `React.lazy` should be wrapped by a `Suspense` component.

Reply:
I ended up wrapping the `Switch` with `Suspense`.

I noticed it in the documentation:
https://reactjs.org/docs/code-splitting.html#route-based-code-splitting